src/main/java/com/teamacronymcoders/epos/feature/quiver/QuiverItem.java

```diff
@@ -0,0 +1,56 @@
+package com.teamacronymcoders.epos.feature.quiver;
+
+import com.hrznstudio.titanium.block.tile.inventory.PosInvHandler;
+import com.teamacronymcoders.epos.Epos;
+import com.teamacronymcoders.epos.api.EposAPI;
+import com.teamacronymcoders.epos.client.gui.GuiQuiverAddonScreen;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.*;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.tags.ItemTags;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Hand;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.event.entity.player.ArrowNockEvent;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+
+import javax.annotation.Nullable;
+
+public class QuiverItem extends Item {
+    public QuiverItem() {
+        super(new Item.Properties().group(Epos.EPOS_TAB).maxStackSize(1).rarity(Rarity.EPIC));
+    }
+
+    @Nullable
+    @Override
+    public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundNBT nbt) {
+        return new QuiverCapabilityProvider(new PosInvHandler("", 0, 0, 9)
+                .setInputFilter((idStack, integer) -> idStack.getItem().isIn(ItemTags.ARROWS) || idStack.getItem() instanceof FireworkRocketItem).setRange(3, 3));
```

Comment:
Can just use the ShootingItem Predicate and call check on it. instead of manually doing the full check.